state/subnets.go

```diff
@@ -175,6 +175,12 @@ func (s *Subnet) AvailabilityZone() string {
 // SpaceName returns the space the subnet is associated with. If the subnet is
 // not associated with a space it will be the empty string.
 func (s *Subnet) SpaceName() string {
+	if s.doc.FanLocalUnderlay != "" {
+		underlay, err := s.st.Subnet(s.doc.FanLocalUnderlay)
+		if err == nil {
+			return underlay.SpaceName()
+		}
```

Comment:
this ends up suppressing all sorts of errors (disconnected mongo, etc), which is probably not great.
I don't really love having SpaceName have a side effect of reading from the Database, though maybe that's just what we have to live with. It may be that most of the time we read a subnet we'll want its space, I suppose waiting until we know we do might be better than double reads always.
Do we know if we have the correct index on subnet.CIDR (I'm guessing we don't)
you'd check in state/allcollections.go 

Thoughts on doing the 2 queries on every Subnet() that is a fan so we don't have to change SpaceName to possibly return an err?
Or alternatively, we do it differently and still always record the space name, but just force that we keep them in sync. so updating the underlay causes the fan entries to get updated.

Reply:
1. I'd do 2 queries in Subnet() 
2. Since FAN subnet space is -always- the space of underlying network we'd have two points of information that could possibly get out of sync, and since it's not a query we need to optimize denormalizing it is IMHO just asking for troubles.

Reply:
fine with me.